src/screens/Editor/Editor.js

```diff
@@ -25,11 +24,15 @@ function Editor ({ match }) {
   const store = React.useContext(AppContext)
   const editorBox = React.useRef(null)
 
-  const params = match.params
+  React.useEffect(() => {
```

Comment:
So if I'm understanding this right, `React.useEffect()` is the equivalent of `componentDidUpdate` and `componentDidMount`.

I can see that this makes sense in the `componentDidMount` way - the resources are fetched as soon as the edit page [(example)](http://localhost:3000/projects/1764/workflows/11/groups/TEST1/subjects/51842/edit) is loaded.

What I'm curious about is that, shouldn't this also trigger when _observed values change?_ (And the Editor component is definitely being `observer()`-ed)

I can't seem to get this to happen though, so there seems to be no problem... I need to read up on `React.useEffect()` more

Reply:
Rerendering is dependent on which items are passed into the array at the end of the function, here being `[match, store]`. Or, you can pass in an empty array instead to say "only trigger once after render." 

However, since `match` and `store` are dependencies here, they are required to be passed into the array.

Check out the final yellow "note" section at the end of this page https://reactjs.org/docs/hooks-effect.html

Reply:
👍